homeassistant/helpers/trigger_template_entity.py

```diff
@@ -176,35 +200,91 @@ def restore_attributes(self, last_state: State) -> None:
-                rendered[key] = self._config[key].async_render(
-                    variables,
-                    parse_result=key in self._parse_result,
-                )
+            return self._config[key].async_render(
+                variables, parse_result=key in self._parse_result, strict=strict
+            )
+        except TemplateError as err:
+            log_triggered_template_error(self.entity_id, err, key=key)
 
-            for key in self._to_render_complex:
-                rendered[key] = render_complex(
-                    self._config[key],
-                    variables,
-                )
+        return _SENTINEL
 
-            if CONF_ATTRIBUTES in self._config:
-                rendered[CONF_ATTRIBUTES] = render_complex(
-                    self._config[CONF_ATTRIBUTES],
-                    variables,
+    def _render_templates(self, variables: dict[str, Any]) -> None:
+        """Render templates."""
+        rendered = dict(self._static_rendered)
+
+        # Check availability first and render as a simple template because
+        # availability should only be able to render True, False, or None.
+        available = True
+        if CONF_AVAILABILITY in self._to_render_simple:
+            # Render availability with strict=True to catch template errors.
+            # This is to ensure feature parity between template entities and trigger
+            # based template entities.  Template entities do not go unavailable when
+            # the availability template renders an error.
+            if (
+                result := self._render_single_template(
+                    CONF_AVAILABILITY, variables, strict=True
                 )
+            ) is not _SENTINEL:
+                rendered[CONF_AVAILABILITY] = available = result
 
+        if not available:
             self._rendered = rendered
-        except TemplateError as err:
-            logging.getLogger(f"{__package__}.{self.entity_id.split('.')[0]}").error(
-                "Error rendering %s template for %s: %s", key, self.entity_id, err
-            )
-            self._rendered = self._static_rendered
+            return
+
+        # If state fails to render, the entity should go unavailable.  Render the
+        # state as a simple template because the result should always be a string or None.
+        if CONF_STATE in self._to_render_simple:
+            if (
+                result := self._render_single_template(CONF_STATE, variables)
+            ) is _SENTINEL:
+                self._rendered = self._static_rendered
+                return
+
+            rendered[CONF_STATE] = result
+        elif CONF_VALUE_TEMPLATE in self._to_render_simple:
+            if (
+                result := self._render_single_template(CONF_VALUE_TEMPLATE, variables)
+            ) is _SENTINEL:
+                self._rendered = self._static_rendered
+                return
+
+            rendered[CONF_VALUE_TEMPLATE] = result
```

Comment:
I don't understand when this happens. Where is `CONF_VALUE_TEMPLATE` added to `_to_render_simple`?

Reply:
Oh, it's a really odd implementation.  It's added when you inherit from the class.

```
class MyNewTemplateClass(TemplateTriggerEntity):
    
        extra_template_keys = (CONF_VALUE_TEMPLATE,)

        ...
```

Reply:
extra_template_keys adds keys it to _to_render_simple.

Reply:
Yes, I know that, but there's no derived class using `CONF_VALUE_TEMPLATE` for the state template AFAICT?

Also, to make the code clearer, maybe we should have a `state_template_key` in the base class, which can be set by derived classes?